nss/integration-tests/integration_test.go

```diff
@@ -153,23 +155,22 @@ func TestIntegration(t *testing.T) {
 }
 
 func TestMain(m *testing.M) {
-	// Build the NSS library and executable in a temporary directory and allow linking to it.
-	tmpDir, cleanup, err := createTempDir()
-	if err != nil {
-		os.Exit(1)
-	}
-	defer cleanup()
-
-	targetDir, libPath = tmpDir, filepath.Join(tmpDir, "libnss_aad.so.2")
-	// Builds the NSS Library.
-	if err = buildRustNSSLib(); err != nil {
-		cleanup()
-		fmt.Fprintf(os.Stderr, "Can not build rust nss library: %v", err)
-		os.Exit(1)
-	}
-
 	testutils.InstallUpdateFlag()
 	flag.Parse()
 
+	if testutils.WantCoverage() {
+		defer func() {
+			// Gets the path to the integration-tests.
+			_, p, _, _ := runtime.Caller(0)
+			currDir := filepath.Dir(p)
+			// Purge .profraw files
+			files, _ := filepath.Glob(filepath.Join(currDir, "*.profraw"))
+			for _, f := range files {
+				os.Remove(f)
+			}
+		}()
+	}
+
 	m.Run()
```

Comment:
ah, here is an issue, you should capture the return value of m.Run(), and error out with the correct exit code after collecting the coverage.

You should find some examples in other projects (or even this one), for this.

Here, if the test fails, you will always return success (and yes, it’s already there and I should have spotted it before, sorry)

Reply:
Well, we should probably also fix adsys then 😂